README.md

```diff
@@ -133,27 +154,41 @@ When accessing the data, you can choose between two internal formats:
+Finally, it takes the "entry types" (`-t`) and creates a public API function for each of them.
+While doing these things, it will make a number of optimizations, e.g. inlining code for small types and removing ununsed functions.
+It outputs the generated code into header and source files and optionally creates a Cmake file to build them.
 
-CDDL allows placing restrictions on the members of your data structure.
-Restrictions can be on type, on content (e.g. values/sizes of ints or strings), and repetition (e.g. the number of members in a list).
-The generated code will validate the input (i.e. the structure if encoding, or the payload for decoding), which means that it will check all the restriction set in the CDDL description, and fail if a restriction is broken.
+So, the `zcbor code` command reads one or more CDDL file(s) and generates some or all of these files:
+ - A header file with types (always)
+ - A header file with declarations for decoding functions (if `--decode`/`-d` is specified)
+ - A C file with decoding functions (if `--decode`/`-d` is specified)
+ - A header file with declarations for encoding functions (if `--encode`/`-e` is specified)
+ - A C file with encoding functions (if `--encode`/`-e` is specified)
+ - A Cmake file that creates a library with the generated code and the C library (if `--output-cmake` is specified).
 
-The cbor libraries do most of the actual translation and moving of bytes, and the validation of values.
+CDDL allows placing restrictions on the members of your data.
+Restrictions can be on type (int/string/list/bool etc.), on content (e.g. values/sizes of ints or strings), and repetition (e.g. the number of members in a list).
+The generated code will validate the input, which means that it will check all the restriction set in the CDDL description, and fail if a restriction is broken.
 
-There are tests for the code generation in [tests/](tests/).
-The tests require [Zephyr](https://github.com/zephyrproject-rtos/zephyr) (if your shell is set up to build Zephyr samples, the tests should also build).
+There are tests for the code generation in [tests/decode](tests/decode) and [tests/encode](tests/encode).
+The tests require [Zephyr](https://github.com/zephyrproject-rtos/zephyr) (if your system is set up to build Zephyr samples, the tests should also build).
 
 Build system
 ------------
 
-When calling zcbor with the argument `--output-cmake <file path>`, a cmake file will be created at that location.
-The cmake file creates a cmake target and adds the generated and non-generated source files, and the include directories to the header files.
-This cmake file can then be included in your project's `CMakeLists.txt` file, and the target can be linked into your project.
-This is demonstrated in the tests, e.g. at tests/decode/test3_simple/CMakeLists.txt.
+When calling zcbor with the argument `--output-cmake <file path>`, a Cmake file will be created at that location.
```

Comment:
```suggestion
When calling zcbor with the argument `--output-cmake <file path>`, a CMake file will be created at that location.
```
The proper name is CMake, I guess we should stick with that? Several places in this doc.